Bottom: 3a40661e5f8d2b112707dba8ccea8740d28b1be2
Top:    7e6828d9a4dbe3608aff84244997f42594963890
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2013-01-01 23:43:00 +0000

hid/gcode: Add a more direct "#include <locale.h>" for setlocale()

The lack of this include caused a failure to build on my machine when
compiled without optimisation (-O0). With any optimisation turned on,
the build would work without the include.

Having chased this strangeness down, it turns out that we were including
libintl.h via global.h, and that libintl.h includes locale.h, IFF
optimisation is turned on at the time it is pre-processed.


---

diff --git a/src/hid/gcode/gcode.c b/src/hid/gcode/gcode.c
index 9a0894f..208d02d 100644
--- a/src/hid/gcode/gcode.c
+++ b/src/hid/gcode/gcode.c
@@ -45,6 +45,10 @@
 
 #include <time.h>
 
+#ifdef HAVE_LOCALE_H
+#include <locale.h>
+#endif
+
 #include "global.h"
 #include "error.h" /* Message() */
 #include "data.h"
